test/cppsuite/test_harness/workload/random_generator.cxx

```diff
@@ -44,10 +44,10 @@ random_generator::instance()
 std::string
 random_generator::generate_random_string(std::size_t length, characters_type type)
 {
-    std::string str;
+    std::string str = get_characters(type);
```

Comment:
what is this?

Reply:
"Small code improvement when generating a random string" from b1004b4c5997e9d83b4a152ddd1dd086228f091a.
It is not related to this change, but when I used the random generator in this test, I realized this code could be improved. Instead of calling `get_characters` in the `while` loop until having a string that is long enough, we can duplicate the string directly by appending the string to itself.

Reply:
`get_characters` will return either the alphabet or some pseudo alphanum characters. But if you need a string of length 10K, we used to gall `get_characters` until having the desired length. But calling `get_characters` implies a function call, a switch statement, a string construction, etc etc.

Reply:
👍

Reply:
This change is fine, but there's a flip side. The code can allocate up to `length - 1` extra bytes of memory.

Reply:
Hmm, how? With the new code changes, `str` has the length of `get_characters(type)` and we duplicate `str`.
Before, we were calling `get_characters(type)` in a loop. These two result in the same memory allocation or am I missing something?

Reply:
As an example, let's say `get_characters(type)` returns 25 characters long string. And we need to generate 401 characters long string. So the while loop iterations will be like this:
1. 25 < 401 ?; str length is 50
2. 50 < 401 ?; str length is 100
3. 100 < 401 ?; str length is 200
4. 200 < 401 ?; str length is 400
5. 400 < 401 ?; str length is **800**. 399 bytes extra allocation. I understand this is more complicated because we use std::string with its internal memory allocation algorithm, but in the worst-case scenario, we can get that result.

The longer the requested string the bigger the memory overhead. While the original algorithm's overhead would be up to 24 bytes in any case.

Reply:
You are totally right, my bad. I was not taking into account that my string was growing and I was duplicating it... sorry!

Reply:
It's all good, it's most likely not a big deal in terms of the real performace and we can do it either way.

Reply:
I think it's better in the long term if we try to do things right from now!
The latest code should get back to previous possible memory overhead without calling `get_characters` in a loop, see 9df472fe81e6c0281a80a4c820407855ce910059.

Reply:
Looks good!